### DISCUSSION TITLE
Slip system angle

### DISCUSSION THREAD

**Discussion by lilarou** [mtex-toolbox/mtex#1726] [Ask Anything] [answered]
Hi MTEX community, 

 I am curently using MTEX for slip system activation. I'd like to know the angle between the X axis and the trace of a given slip system for a grain orientation. 
For exemple, I with the following code I have a pyramidal slip system with 6 variant for a Ti sample. 

```
ID = 107;
grainID= grains(ID);
sSHexPy2a =slipSystem.pyramidal2CA(CS_Ti);
sSHex2 = symmetrise(sSHexPy2a,'antipodal');
sSID = grainID.meanOrientation* sSHex2;

angle2=angle(sSID.trace, vector3d.X)*180/pi;

figure
plot(grainID,grainID.meanOrientation)
hold on
plot(grainID.boundary,'lineWidth',2)
hold on
for i=1:length(sSID)
    quiver(grainID,sSID(i).trace)
end
legend
hold off
```

With this code I obtain the following figure. The _angle2_ list gives the following angle : [37.03	63.43 66.61 29.10 18.11 23.51]
All angles are given between [0;90]degree but as you can observe the trace of data 3, data 4 and date 6 those 3 angles should be "180-angle" or "-angle".
It looks like the code does not take into account the direction of the X vector and gives the min angle calculated. I would like to get the corresponding angle with X vector in the positive direction (to the right).

![slip_system_trace](https://user-images.githubusercontent.com/123663516/215055613-b5d34f1a-d666-4a11-8f2f-97282c1dbd32.jpg)

Thank you !

**Comment by kilir** [comment, marked as answer]
Hi, 
since sS.trace returns an antipodal vector, the angle between this vector and e.g. an xvector will be within 0:pi. Maybe you'd like to look at the direction of the trace ` mod(sSID.trace.rho,pi)/degree`?
Cheers,
Rüdiger

**Reply by lilarou** [reply]
Thank you Rudiger, this is exactly what I was looking for !
